AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -245,6 +263,28 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
                 self.log.verbose("mutationEvent deleted successfully")
             }
 
+            let localModel: Model
+            do {
+                localModel = try mutationEvent.decodeModel()
+            } catch {
+                self.log.error("Couldn't decode local model")
+                self.stateMachine.notify(action: .processedEvent)
+                return
+            }
+
+            /// Only emit `outboxMutationProcessed` when a successful response returns
+            guard let mutationSyncMetadata = mutationSyncMetadata else {
+                self.stateMachine.notify(action: .processedEvent)
+                return
+            }
+            let outboxMutationProcessedEvent = OutboxMutationEvent
+                .fromModelWithMetadata(modelName: mutationEvent.modelName,
+                                       model: localModel,
+                                       mutationSync: mutationSyncMetadata)
```

Comment:
looks like some of this can be refactored
```
if let mutationSyncMetadata = mutationSyncMetadata {
  self.dispatchOutboxMutationEvent(mutationEvent, mutationSyncMetadata)
} 

self.queryMutationEventsFromStorage {
   self.stateMachine.notify(action: .processedEvent)
}
```

```
private func dispatchOutboxMutationEvent(mutationEvent, mutationSyncMetadata) {
  do {
    // localModel
   // create outBoxMutationProcessedEvent
   // create payload
   // Amplify.dispatch 
  } catch {
   //log
    return
  }


```